workflow-ui/compose/src/androidTest/java/com/squareup/workflow1/ui/compose/RenderAsStateTest.kt

```diff
@@ -212,6 +225,107 @@ class RenderAsStateTest {
     }
   }
 
+  @Test fun renderingIsAvailableImmediatelyWhenWorkflowScopeUsesDifferentDispatcher() {
+    val workflow = Workflow.rendering("hello")
+    val scope = TestCoroutineScope()
+
+    // Ensure the workflow runtime won't actually run aside from the synchronous first pass.
+    scope.pauseDispatcher()
+
+    composeRule.setContent {
+      val initialRendering = workflow.renderAsState(
+        props = Unit, onOutput = {},
+        scope = scope
+      )
+      assertThat(initialRendering.value).isNotNull()
+    }
+
+    scope.cleanupTestCoroutines()
```

Comment:
This cleanup wont happen if the test fails on an earlier line. Same for tests below.

If you don't want to set up the scope with `@Before`/`@After`, one pattern I've used in tests before for clean up is `try`/`finally`

Reply:
I think if the test fails this scope is already going to have been torn down? That might not be true though and it's better not to rely on it. 

- [ ] Fix coroutines cleanup